File: backend/app/schemas/etf.py
from datetime import date

from pydantic import BaseModel


class PriceHistoryPointResponse(BaseModel):
    date: date
    open: float | None
    high: float | None
    low: float | None
    close: float | None
    volume: int | None


class ETFPriceHistoryResponse(BaseModel):
    symbol: str
    prices: list[PriceHistoryPointResponse]

File: frontend/src/types/etf.ts
export interface PriceHistoryPoint {
  date: string;
  open: number | null;
  high: number | null;
  low: number | null;
  close: number | null;
  volume: number | null;
}

export interface ETFPriceHistoryResponse {
  symbol: string;
  prices: PriceHistoryPoint[];
}
